lib/active_model_serializers/adapter/base.rb

```diff
@@ -78,6 +83,32 @@ def serialization_options(options)
       def root
         serializer.json_key.to_sym if serializer.json_key
       end
+
+      def fields_to_hash(fields)
+        case fields
+        when Hash
+          fields
+        when Array
```

Comment:
To follow the ruby philosophy, I think it would be of a greater benefit if we used `respond_to?`. This way you can send in object that implement `Hash` or `Array`, but aren't necessarily those objects themselves.